### PR TITLE
Add new react implementation using redux, immutable.js

### PR DESCRIPTION
**Pull request by taehwanno**
This is another example of HNPWA using React. But I use redux and immutable.js for state management. PTAL 😄 

Github Repository: https://github.com/taehwanno/hnpwa-react
Running Application: https://hnpwa-react.firebaseapp.com/news/1

### CODE REVIEW

File: site/_apps/hnpwa-react.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+perfomance-patterns: 'Server-side data pre-fetching, preload/prefetch resources'
+server-side-rendering: 'Yes'
+api: 'Node-hnapi (unofficial)'
+hosting: 'Firebase'
+authors:
+    - name: 'Taehwan, No'
+lighthouse: '91/100'
+interactive-em: '5.2s'
+interactive-faster-3g: '4.3s'
+lighthouse-link: https://www.webpagetest.org/lighthouse.php?test=171029_69_655c5b28865ee880316d1af49358b1a0&run=1
```
Comment (addyosmani) [inline site/_apps/hnpwa-react.txt]: Performance metrics look about right:

image

I'll check WebPageTest next.

File: site/_apps/hnpwa-react.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+wpt-em-link: https://www.webpagetest.org/result/171029_6Y_04df9e72abb38731d412b0eeb7be1317/
+wpt-faster-3g-link: https://www.webpagetest.org/result/171029_69_655c5b28865ee880316d1af49358b1a0/
```
Comment (addyosmani) [inline site/_apps/hnpwa-react.txt]: JavaScript settles on the main thread in under 5 seconds here. All good:

image

**Comment by addyosmani** [comment]
Next up is the UX review.

* Verified routes are returning the correct data
* Verified navigating to shared URLs for a route load up the correct page
* Prev/Next navigation in the UI is as expected
* Offline data fetching from the cache is working as expected
* Threaded comments appear to be rendering well

**Comment by addyosmani** [comment]
Friendly ping just to check if you're still interested in pursuing this implementation :)

**Comment by taehwanno** [comment]
Oh, Sure. I'm waiting for your response :) 👍

**Comment by addyosmani** [comment]
@taehwanno Would it be possible to adjust the pagination UX in your implementation to it shows the current/total page numbers? See https://react-hn.kristoferbaxter.com/ for an example. In between Prev/Next you'll see the 1/12 -> 2/12 etc as you navigate through.

**Comment by taehwanno** [comment]
@addyosmani OK 👍  It's done at https://github.com/taehwanno/hnpwa-react/pull/21.

**Comment by taehwanno** [comment]
@housseindjirdeh Oh, Thanks. 😄 🙏 🚀

**Pull request merged**